src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs

```diff
@@ -23,107 +22,305 @@ internal static class TrustedSignersCommand
-                    {
-                        action = TrustCommand.List;
-                    }
-                    else if (!Enum.TryParse(command.Values[0], ignoreCase: true, result: out action))
-                    {
-                        throw new CommandLineArgumentCombinationException(string.Format(CultureInfo.CurrentCulture, Strings.Error_UnknownAction, command.Values[0]));
-                    }
+        private static async Task<int> ExecuteCommand(TrustCommand action,
+                      CommandOption algorithm,
+                      bool allowUntrustedRootOption,
+                      CommandOption owners,
+                      CommandOption verbosity,
+                      CommandOption configFile,
+                      Func<ILogger> getLogger,
+                      Action<LogLevel> setLogLevel,
+                      string name = null,
+                      string sourceUrl = null,
+                      string packagePath = null,
+                      string fingerprint = null)
+        {
+            ILogger logger = getLogger();
 
-                    string name = null;
+            try
+            {
+                ISettings settings = ProcessConfigFile(configFile.Value());
 
-                    if (command.Values.Count > 1)
-                    {
-                        name = command.Values[1];
-                    }
+                var trustedSignersArgs = new TrustedSignersArgs()
+                {
+                    Action = MapTrustEnumAction(action),
+                    PackagePath = packagePath,
+                    Name = name,
+                    ServiceIndex = sourceUrl,
+                    CertificateFingerprint = fingerprint,
+                    FingerprintAlgorithm = algorithm?.Value(),
+                    AllowUntrustedRoot = allowUntrustedRootOption,
+                    Author = action == TrustCommand.Author,
+                    Repository = action == TrustCommand.Repository,
+                    Owners = CommandLineUtility.SplitAndJoinAcrossMultipleValues(owners?.Values),
+                    Logger = logger
+                };
 
-                    string packagePath = null;
-                    string sourceUrl = null;
-                    string fingerprint = null;
-                    if (command.Values.Count() > 2)
+                setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var sourceProvider = new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false);
+#pragma warning restore CS0618 // Type or member is obsolete
+                var trustedSignersProvider = new TrustedSignersProvider(settings);
+
+                var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                Task<int> trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                return await trustedSignTask;
```

Comment:
``` suggestion
                return trustedSignTask;
```

Reply:
Even this suggestion saves 1 machine state, but makes hard to track down aggregated exception(I couldn't find the link for ref, it brought up during last year learning day for async programming), maybe better to keep as it's since all other `dotnet nuget` implementation doing same.